### PR TITLE
added AuthenticateUser()  and GetUserLicenses()

### PR DESCRIPTION
**Pull request by ahmad-kemsan**

### CODE REVIEW

File: src/lexactivator-exception.ts [hunk 1]
```diff
@@ -204,6 +204,10 @@ export class LexActivatorException extends Error {
 			message = 'Release channel not set.';
 			break;
 
+		case LexStatusCodes.LA_E_USER_NOT_AUTHENTICATED:
+			message = 'Permission Denied.';
```
Comment (adnan-kamili) [inline src/lexactivator-exception.ts:208]: which permission is denied?

File: src/lexactivator.ts [hunk 2]
```diff
@@ -71,6 +71,15 @@ export class OrganizationAddress {
...
 };
 
+export class UserLicense {
+	allowedActivations: number;
+	allowedDeactivations: number;
+	key: string;
+	type: string;
+}
+
+export type UserLicenses = UserLicense[];
```
Comment (adnan-kamili) [inline src/lexactivator.ts:81]: why do need to create a type for an array?

File: src/lexstatus-codes.ts [hunk 3]
```diff
@@ -298,6 +298,11 @@ export const LexStatusCodes = {
 	 */
 	LA_E_RELEASE_CHANNEL: 86,
 
+	/**
+	 * Permission denied.
```
Comment (adnan-kamili) [inline src/lexstatus-codes.ts:302]: whihc permission is not denied

File: src/native/LexActivator.h [hunk 4]
```diff
@@ -823,6 +843,16 @@ LEXACTIVATOR_API int LA_CC CheckReleaseUpdateInternal(ReleaseCallbackType releas
...
 
+/*
+    FUNCTION: AuthenticateUser()
+
+    PURPOSE: It sends the request to the Cryptlex servers to authenticate the user.
+
+    RETURN CODES: LA_OK, LA_E_PRODUCT_ID, LA_E_INET, LA_E_SERVER, LA_E_RATE_LIMIT
+    LA_E_AUTHENTICATION_FAILED
+*/
+LEXACTIVATOR_API int LA_CC AuthenticateUser(CSTRTYPE email, CSTRTYPE password);
```
Comment (adnan-kamili) [inline src/native/LexActivator.h:854]: parameters missing

File: src/native/LexStatusCodes.h [hunk 5]
```diff
@@ -422,6 +422,13 @@ enum LexStatusCodes
     */
     LA_E_RELEASE_CHANNEL = 86,
 
+    /*
+        CODE: LA_E_USER_NOT_AUTHENTICATED
+
+        MESSAGE: Permission denied.
```
Comment (adnan-kamili) [inline src/native/LexStatusCodes.h:428]: please check

**Review by adnan-kamili — CHANGES_REQUESTED** [review]
please mark pull request ready for review if ready for review

File: src/lexactivator.ts (hunk 2 above)
Comment (adnan-kamili) [reply src/lexactivator.ts:81]: why would we create types for an array? Have you seen this behaviour anywhere in any library?

Comment (ahmad-kemsan) [reply src/lexactivator.ts:81]: No, it's not something standard or recommended way. It just felt clean to me but now I realize it's better to keep it UserLicense[]

**Review by adnan-kamili — APPROVED** [review]

**Pull request merged**